internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,637 @@
+	}
+
+	vpcs, err := d.client.GetVpcsWithFilters([]*ec2.Filter{
+		{
+			Name:   aws.String(VpcClusterIDTagKey),
+			Values: []*string{aws.String(clusterInstallations[0].ClusterID)},
+		},
+		{
+			Name:   aws.String(VpcAvailableTagKey),
+			Values: []*string{aws.String(VpcAvailableTagValueFalse)},
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to lookup VPCs for installation ID %s", d.installationID)
+	}
+	if len(vpcs) != 1 {
+		return nil, fmt.Errorf("expected 1 VPC for RDS cluster %s (found %d)", clusterInstallations[0].ClusterID, len(vpcs))
+	}
+
+	return vpcs[0], nil
+}
+
+func (d *RDSMultitenantDatabase) updateCounterTag(resourceARN *string, counter int) error {
+	_, err := d.client.Service().rds.AddTagsToResource(&rds.AddTagsToResourceInput{
+		ResourceName: resourceARN,
+		Tags: []*rds.Tag{
+			{
+				Key:   aws.String(trimTagPrefix(DefaultMultitenantDatabaseCounterTagKey)),
+				Value: aws.String(fmt.Sprintf("%d", counter)),
+			},
+		},
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to update RDS cluster counter's tag")
+	}
+
+	return nil
+}
+
+func (d *RDSMultitenantDatabase) createInstallationSecret(secretName, username, description string, tags []*secretsmanager.Tag) (*RDSSecret, error) {
+	rdsSecretPayload := RDSSecret{
+		MasterUsername: username,
+		MasterPassword: newRandomPassword(40),
+	}
+	err := rdsSecretPayload.Validate()
+	if err != nil {
+		return nil, errors.Wrapf(err, "secret %s has error(s)", secretName)
+	}
+
+	b, err := json.Marshal(&rdsSecretPayload)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to marshal secrets manager payload")
+	}
+
+	_, err = d.client.Service().secretsManager.CreateSecret(&secretsmanager.CreateSecretInput{
+		Name:         aws.String(secretName),
+		Description:  aws.String(description),
+		Tags:         tags,
+		SecretString: aws.String(string(b)),
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create secret %s", secretName)
+	}
+
+	return &rdsSecretPayload, nil
+}
+
+func (d *RDSMultitenantDatabase) describeRDSCluster(dbClusterID string) (*rds.DBCluster, error) {
+	dbClusterOutput, err := d.client.Service().rds.DescribeDBClusters(&rds.DescribeDBClustersInput{
+		Filters: []*rds.Filter{
+			{
+				Name:   aws.String("db-cluster-id"),
+				Values: []*string{aws.String(dbClusterID)},
+			},
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get RDS cluster id %s", dbClusterID)
+	}
+	if len(dbClusterOutput.DBClusters) != 1 {
+		return nil, fmt.Errorf("expected exactly one RDS cluster for installation id %s (found %d)", d.installationID, len(dbClusterOutput.DBClusters))
+	}
+
+	return dbClusterOutput.DBClusters[0], nil
+}
+
+func (d *RDSMultitenantDatabase) acquireLock(dbClusterID string, store model.InstallationDatabaseStoreInterface) (func(logger log.FieldLogger), error) {
```

Comment:
The naming here is a little confusing. A comment would help enlighten me as to the intended difference between this and `lockRDSCluster` but a more specific name for one or both of these methods would improve this. As it is, `myRDSMultitenantDatabase.lockRDSCluster()` and `myRDSMultitenantDatabase.acquireLock()` sound like they should do the same thing.